CMakeLists.txt

```diff
@@ -1039,713 +1039,6 @@ if(OPENCV_GENERATE_SETUPVARS)
-  status("")
-  status("  Python 2:")
-  status("    Interpreter:"     PYTHON2INTERP_FOUND  THEN "${PYTHON2_EXECUTABLE} (ver ${PYTHON2_VERSION_STRING})"       ELSE NO)
-  if(PYTHON2LIBS_VERSION_STRING)
-    status("    Libraries:"   HAVE_opencv_python2  THEN  "${PYTHON2_LIBRARIES} (ver ${PYTHON2LIBS_VERSION_STRING})"   ELSE NO)
-  else()
-    status("    Libraries:"   HAVE_opencv_python2  THEN  "${PYTHON2_LIBRARIES}"                                      ELSE NO)
-  endif()
-  status("    numpy:"         PYTHON2_NUMPY_INCLUDE_DIRS THEN "${PYTHON2_NUMPY_INCLUDE_DIRS} (ver ${PYTHON2_NUMPY_VERSION})" ELSE "NO (Python wrappers can not be generated)")
-  status("    install path:"  HAVE_opencv_python2  THEN "${__INSTALL_PATH_PYTHON2}"                            ELSE "-")
-endif()
-
-if(BUILD_opencv_python3)
-  status("")
-  status("  Python 3:")
-  status("    Interpreter:"     PYTHON3INTERP_FOUND  THEN "${PYTHON3_EXECUTABLE} (ver ${PYTHON3_VERSION_STRING})"       ELSE NO)
-  if(PYTHON3LIBS_VERSION_STRING)
-    status("    Libraries:"   HAVE_opencv_python3  THEN  "${PYTHON3_LIBRARIES} (ver ${PYTHON3LIBS_VERSION_STRING})"   ELSE NO)
-  else()
-    status("    Libraries:"   HAVE_opencv_python3  THEN  "${PYTHON3_LIBRARIES}"                                      ELSE NO)
-  endif()
-  status("    numpy:"         PYTHON3_NUMPY_INCLUDE_DIRS THEN "${PYTHON3_NUMPY_INCLUDE_DIRS} (ver ${PYTHON3_NUMPY_VERSION})" ELSE "NO (Python3 wrappers can not be generated)")
-  status("    install path:"  HAVE_opencv_python3  THEN "${__INSTALL_PATH_PYTHON3}"                            ELSE "-")
-endif()
-
-status("")
-status("  Python (for build):"  PYTHON_DEFAULT_AVAILABLE THEN "${PYTHON_DEFAULT_EXECUTABLE}" ELSE NO)
-
-# ========================== java ==========================
-if(BUILD_JAVA)
-  status("")
-  status("  Java:"            BUILD_FAT_JAVA_LIB  THEN "export all functions"                                      ELSE "")
-  status("    ant:"           ANT_EXECUTABLE      THEN "${ANT_EXECUTABLE} (ver ${ANT_VERSION})"                    ELSE NO)
-  if(NOT ANDROID)
-    status("    JNI:"         JNI_INCLUDE_DIRS    THEN "${JNI_INCLUDE_DIRS}"                                       ELSE NO)
-  endif()
-  status("    Java wrappers:" HAVE_opencv_java                                                            THEN YES ELSE NO)
-  status("    Java tests:"    BUILD_TESTS AND opencv_test_java_BINARY_DIR                                 THEN YES ELSE NO)
-endif()
-
-# ========================== Objective-C =======================
-if(BUILD_OBJC)
-  status("")
-  status("  Objective-C wrappers:" HAVE_opencv_objc                                                       THEN YES ELSE NO)
-endif()
-
-ocv_cmake_hook(STATUS_DUMP_EXTRA)
-
-# ========================== auxiliary ==========================
-status("")
-status("  Install to:" "${CMAKE_INSTALL_PREFIX}")
-status("-----------------------------------------------------------------")
-status("")
-
-
-ocv_finalize_status()
```

Comment:
I'd say this cmake output is valuable and helps debugging build configurations. It should stay. We only need to prevent it to become part of the binary.

Reply:
You need to discover where these status strings get embedded into the C++ code and/or binary.